File: povoamento_srag.py
import os
import time
import pandas as pd
from dotenv import load_dotenv
from sqlalchemy import  create_engine
from urllib.parse import quote_plus as urlquote
load_dotenv()
inicio = time.time()
try:
    print("Iniciando conexão")
    engine = create_engine(url="postgresql://%s:%s@%s" % (os.getenv('USER'), urlquote(os.getenv('PASSWORD')),
                                    os.getenv('HOST') + ":5432/" + os.getenv('DB_NAME')), echo = False ).connect()
    print("Conexão feita")
    print("Montando tabela povoamento_srag")
    dataframe = pd.concat([pd.DataFrame(pd.read_csv("data/"+filename,sep=";",low_memory=False)) for filename in os.listdir("data")])
    dataframe.columns = [col.lower() for col in dataframe.columns]
    dataframe.to_sql('srag', con = engine, index=False,if_exists='replace',schema='public',chunksize=100000)
    print("Fim do processo de povoamento")
except Exception as e:
    print(f'Erro:{e}')
fim = time.time()
print(f"Tempo total: {(fim-inicio)/60}")

File: povoamento_srag.jl
using CSV
using DotEnv
using LibPQ
using DataFrames
DotEnv.load()

function make_dataframe()
    return vcat([ CSV.File("data/$(filename)") |> DataFrame for filename in readdir("data/")]...)
end

println("Iniciando conexão com o banco de dados")
connection = LibPQ.Connection("dbname=$(ENV["DB_NAME"]) user=$(ENV["USER"]) password=$(ENV["PASSWORD"])")
println("Conexão feita com sucesso")

println("Carregando datasets para memoria . . .")
dataframe = make_dataframe()
println("Dataset carregado para memoria")
rename!(dataframe, [Symbol(lowercase(string(col))) for col in names(dataframe)])


#=
println("Enviando DataFrame para a tabela 'srag' no banco de dados...")
query = """CREATE TABLE IF NOT EXISTS public.povoamento_srag(  
);"""
LibPQ.execute(connection, query)
LibPQ.load!(connection, "srag", dataframe; if_exists="append", chunk_size=100000)
println("DataFrame enviado para a tabela 'srag' no banco de dados")

# Fechando a conexão com o banco de dados
LibPQ.finish(connection)
=#